src/miniscript/satisfy.rs

```diff
@@ -25,12 +25,25 @@ use bitcoin::hashes::{hash160, ripemd160, sha256, sha256d};
 use bitcoin::{self, secp256k1};
 use {MiniscriptKey, ToPublicKey};
 
+use Error;
 use ScriptContext;
 use Terminal;
 
 /// Type alias for a signature/hashtype pair
 pub type BitcoinSig = (secp256k1::Signature, bitcoin::SigHashType);
 
+/// Helper function to create BitcoinSig from Rawsig
+/// Useful for downstream when implementing Satisfier.
+/// Returns underlying secp if the Signature is not of correct formart
+pub fn bitcoinsig_from_rawsig(rawsig: &[u8]) -> Result<BitcoinSig, Error> {
+    let (flag, sig) = rawsig.split_last().unwrap();
+    let flag = bitcoin::SigHashType::from_u32(*flag as u32);
+    let sig = match secp256k1::Signature::from_der(sig) {
```

Comment:
Will
```
let sig = secp256k1::Signature::from_der(sig)?;
```
work?

Reply:
Yup, that should be cleaner. We already have a `From` implementation from the `Secp` error